MapAnalyzer/Pather.py

```diff
@@ -81,28 +125,95 @@ def find_all_paths(self, start: Region, goal: Region, path: Optional[List[Region
-        if include_destructables:
-            for rock in self.map_data.bot.destructables:
-                if "plates" not in rock.name.lower():
-                    self.add_cost(position=rock.position.rounded, radius=1 * rock.radius, arr=grid, weight=np.inf, safe=False)
-        return grid
-    
-    def lowest_cost_points_array(self, from_pos: tuple, radius: float, grid: np.ndarray) -> np.ndarray:
+            size = 1
+            if obj.type_id in buildings_2x2:
+                size = 2
+            elif obj.type_id in buildings_3x3:
+                size = 3
+            elif obj.type_id in buildings_5x5:
+                size = 5
+            left_bottom = obj.position.offset((-size / 2, -size / 2))
+            x_start = int(left_bottom[0])
+            y_start = int(left_bottom[1])
+            x_end = int(x_start + size)
+            y_end = int(y_start + size)
+
+            ret_grid[x_start:x_end, y_start:y_end] = 0
+
+            # townhall sized buildings should have their corner spots pathable
+            if size == 5:
+                ret_grid[x_start, y_start] = 1
+                ret_grid[x_start, y_end - 1] = 1
+                ret_grid[x_end - 1, y_start] = 1
+                ret_grid[x_end - 1, y_end - 1] = 1
+
+        if len(self.minerals_included) != self.map_data.bot.mineral_field.amount:
+
+            new_positions = set(m.position for m in self.map_data.bot.mineral_field)
+            old_mf_positions = set(self.minerals_included)
+
+            missing_positions = old_mf_positions - new_positions
+            for mf_position in missing_positions:
+                x1 = int(mf_position[0])
+                x2 = x1 - 1
+                y = int(mf_position[1])
+
+                ret_grid[x1, y] = 1
+                ret_grid[x2, y] = 1
+
+                self.default_grid[x1, y] = 1
+                self.default_grid[x2, y] = 1
+
+                self.default_grid_nodestr[x1, y] = 1
+                self.default_grid_nodestr[x2, y] = 1
+
+                del self.minerals_included[mf_position]
+
+        if include_destructables and len(self.destructables_included) != self.map_data.bot.destructables.amount:
+            new_positions = set(d.position for d in self.map_data.bot.destructables)
+            old_dest_positions = set(self.destructables_included)
+            missing_positions = old_dest_positions - new_positions
+
+            for dest_position in missing_positions:
+                dest = self.destructables_included[dest_position]
+                change_destructable_status_in_grid(ret_grid, dest, 1)
+                change_destructable_status_in_grid(self.default_grid, dest, 1)
+
+                del self.destructables_included[dest_position]
+
+        return ret_grid
+
+    def find_eligible_point(self, point: Tuple[float, float], grid: np.ndarray, terrain_height: np.ndarray, max_distance: float) -> Optional[Tuple[int, int]]:
```

Comment:
This function compares heights to get retreat paths that wont get units stuck, right ?  
i would like to add a doc string describing that here